File: JVM/jnetcall-java/src/main/java/jnetcall/java/server/ServiceLot.java
package jnetcall.java.server;

import jnetcall.java.api.MethodCall;
import jnetproto.java.beans.ProtoConvert;

import java.io.ByteArrayInputStream;
import java.io.OutputStream;
import java.lang.reflect.Method;

public final class ServiceLot<T> extends AbstractHost<T> implements AutoCloseable {
    private T _instance;
    private Method[] _methods;

    public ServiceLot(Class<T> serviceClass) {
        super(serviceClass);
    }

    @Override
    public void close() throws Exception {
        super.close();
    }

    public void build() throws Exception {
        _instance = createInst();
        _methods = _instance.getClass().getMethods();
        ServiceLots.register(this);
    }

    boolean tryCall(byte[] in, OutputStream output) throws Exception {
        try (var input = new ByteArrayInputStream(in);
             var proto = new ProtoConvert(input, output, config)) {
            var call = proto.readObject(MethodCall.class);
            handleCall(_instance, _methods, call, proto);
            output.flush();
            return true;
        }
    }
}

File: NET/JNetCall.Sharp.InProc/Server/ServiceLot.cs
using System;
using System.IO;
using System.Reflection;
using JNetCall.Sharp.API;
using JNetCall.Sharp.API.Flow;
using JNetProto.Sharp.Beans;

namespace JNetCall.Sharp.Server
{
    public sealed class ServiceLot<T> : AbstractHost<T>, ICaller
    {
        private T _instance;
        private MethodInfo[] _methods;

        public ServiceLot(Type serviceClass) : base(serviceClass)
        {
        }

        public void Build()
        {
            _instance = CreateInst();
            _methods = _instance.GetType().GetMethods();
            ServiceLots.Register(this);
        }

        public bool TryCall(byte[] @in, Stream output)
        {
            using var input = new MemoryStream(@in);
            using var proto = new ProtoConvert(input, output, _config);
            var call = proto.ReadObject<MethodCall>();
            HandleCall(_instance, _methods, call, proto);
            output.Flush();
            return true;
        }
    }
}